combiner/combiner.go

```diff
@@ -0,0 +1,208 @@
+package combiner
+
+import (
+	"runtime"
+	"sync"
+
+	"github.com/philippgille/gokv"
+)
+
+// Check we satisfy interface.
+var _ gokv.Store = &Store{}
+
+// Store is a `gokv.Store` implementation that forwards its
+// calls to other `gokv.Store`s.
+type Store struct {
+	backends []gokv.Store
+	parallel bool
+}
+
+// A helper type to check for errors when creating `gokv.Store`s.
+type Backend struct {
```

Comment:
This is only for convenience, right? Maybe it's better to export as few types and functions as possible and the usability is not much worse in this case. A package user probably mostly uses 2 or 3 backends, so it's just a few more lines of code for checking potential errors in those 2 or 3 cases.

This would also allow us to change the `NewStore` factory function to accept variadic parameters. It could look like this:
`func NewStore(options Options, backends ...gokv.Store) (Store, error)`.

Also, it's currently not optional to use this type for convenience, because the current `NewStore` function takes an `Options` argument which only contains `Backend`s instead of `gokv.Store`s. But what the package user's code is structured in a way that there are already initialized, working stores and its known that there are no errors? In this case using `MustBackend()` is required and leads to some extra lines of code.

This also feels like a level of indirection. It's easier for a package user to understand if he can create a combiner directly with multiple stores.

What do you think?